Update Numark Mixtrack Pro Controller

joshpatten opened the pull request:
I know this is an old controller, but I got one extremely cheap from a second hand store, and many of the functions were using deprecated API calls that either gave warnings or did not function at all. As far as I can tell the controller is 100% functional with this XML and JS, but I would like some feedback if someone else can test.

joshpatten committed:
Update Mixtrack Pro Controller

Update the Numark Mixtrack Pro Controller to current mixxx standards

joshpatten committed:
Accidental tabs

joshpatten committed:
Find all tabs

joshpatten committed:
Function Call Fix

joshpatten committed:
Fix Clip Flash

ronso0 commented on res/controllers/Numark Mixtrack Pro.midi.xml:
```diff
@@ -81,30 +81,57 @@ Modify by Darío José Freije 06/27/2012-->
+            <control>
+                <group>[EffectRack1_EffectUnit1]</group>
+                <key>mix</key>
+                <description>MIDI Learned from 569 messages.</description>
```
Please remove lines like this. They'll be displayed in the I/O tables and don't provide useful information.

jpattWPC committed:
Formatting

Variable scope, remove unnecessary XML description, attribution, general formatting.

jpattWPC committed:
Code style issue fix

vespadj commented:
I would also like to improve this mapping.
Last night I applied expanding of folder in the directory mode.
https://mixxx.discourse.group/t/expand-folder-with-browser-folders-files-knob/29172 

Tonight I'll try your mapping and give you some feedback.

vespadj commented:
Hi @joshpatten! in consolle I receive this warnings.
```
warning [Controller] ControlObject accessed via deprecated key "[Master]" "headVolume" - use "[Master]" "headGain" instead

Same warning for:
- "[Channel1]" "filterHigh" - use "[EqualizerRack1_[Channel1]_Effect1]" "parameter3" instead
- "[Channel1]" "filterMid"  - use "[EqualizerRack1_[Channel1]_Effect1]" "parameter2" instead
- "[Channel1]" "filterLow"  - use "[EqualizerRack1_[Channel1]_Effect1]" "parameter1" instead

- "[Channel1]" "filterHigh" - use "[EqualizerRack1_[Channel1]_Effect1]" "parameter3" instead
- "[Channel1]" "filterHigh" - use "[EqualizerRack1_[Channel1]_Effect1]" "parameter3" instead

- "[Channel2]" "filterHigh" - use "[EqualizerRack1_[Channel2]_Effect1]" "parameter3" instead
- "[Channel2]" "filterMid" - use "[EqualizerRack1_[Channel2]_Effect1]" "parameter2" instead
- "[Channel2]" "filterLow" - use "[EqualizerRack1_[Channel2]_Effect1]" "parameter1" instead
```
Mapping works fine and as expected, but "back" Button under Browser, do nothing.
As I already write over, I think on "folder" Browser knob must expand the folder, the previous mapping use "back btn" to switch Folder/File.

My idea:
```js
NumarkMixTrackPro.onCentralKnobPress = function(channel, control, value, status, group) {
	if (!value) return;

	if (NumarkMixTrackPro.directoryMode) {
		engine.setParameter('[Playlist]', 'ToggleSelectedSidebarItem', 1);
	} else {
		engine.setParameter('[Playlist]', 'LoadSelectedIntoFirstStopped', 1);
	}
}
```
diffs screenshot:
![Schermata a 2024-03-12 22-51-49](https://github.com/mixxxdj/mixxx/assets/1891045/5eff67b7-56cd-46fe-8073-18c479375fa1)

ronso0 commented:
FYI you can comment code and make suggestions in the ['Files changed' tab](https://github.com/mixxxdj/mixxx/pull/12948/files). Floating code blocks and screenshots are a bit of ... slow workflow.

And you can always open PRs against @joshpatten's branch https://github.com/joshpatten/mixxx/tree/mixtrack-pro-fix ; )
Or just post commit hashs for others to `git cherry-pick [hash]`.

vespadj commented:
Effects gesture is fine but a little hard for me.
I miss main "Filter", I'll add in some way... but I need to do some test.

vespadj commented:
Sorry @ronso0 , I'm terrible with git :-)
It's a good opportunity to learn.

acolombier commented:
Hey @joshpatten, are you still looking at completing this? I'll make that a draft PR for now, but feel free to mark it ready if it should be reconsidered for merging.
